### PR TITLE
fix: check for pgversion before querying for matviews

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: https://github.com/lightdash/lightdash/issues/11885

### Description:

Previously we were querying `pg_catalog.pg_matviews` without checking first for the PG version.
`pg_matviews` is only available from PostgreSQL >= 12: https://www.postgresql.org/docs/current/view-pg-matviews.html

Logs added for debug purposes:

**PG 11 (doesn't support matviews)**

image

**PG 15 (supports matviews)**
image (1)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: examples/full-jaffle-shop-demo/dbt/models/metrics.yml [hunk 1]
```diff
@@ -1,35 +0,0 @@
-version: 2
```
Comment (rephus) [inline examples/full-jaffle-shop-demo/dbt/models/metrics.yml:1]: could you rollback these  unrelated changes ? not sure if this is still needed somewhere...

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 2]
```diff
@@ -314,6 +314,13 @@ export class PostgresClient<
         if (databases.size <= 0 || schemas.size <= 0 || tables.size <= 0) {
             return {};
         }
+
+        const { rows: supportsMatViewsRows } = await this.runQuery(
+            `SELECT current_setting('server_version_num')::integer >= 120000 as supports_matviews`,
+        );
+        const supportsMatviews =
+            supportsMatViewsRows[0]?.supports_matviews ?? '';
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:322]:
```suggestion
            supportsMatViewsRows[0]?.supports_matviews ?? false;
```

Should this be a boolean ? rather than an empty string

**Review by rephus — APPROVED** [review]

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 3]
```diff
@@ -314,6 +314,13 @@ export class PostgresClient<
         if (databases.size <= 0 || schemas.size <= 0 || tables.size <= 0) {
             return {};
         }
+
+        const { rows: supportsMatViewsRows } = await this.runQuery(
+            `SELECT current_setting('server_version_num')::integer >= 120000 as supports_matviews`,
```
Comment (joaoviana) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:319]: so this is >= v12?

Comment (almeidabbm) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:319]: yep yep, `pg_matviews` is only available on and after 12: https://www.postgresql.org/docs/current/view-pg-matviews.html
